docs/upgrade.md

```diff
@@ -0,0 +1,93 @@
+### Starting conditions:
+- A compatible seed image with target cluster is created. Seed image has `4.16.y` ocp version. [Both target cluster and seed image has configured a shared container partition for IBU.](https://docs.openshift.com/container-platform/4.17/edge_computing/image_based_upgrade/preparing_for_image_based_upgrade/cnf-image-based-upgrade-shared-container-partition.html)
+- Spoke cluster is successfully deployed and configured using a `ProvisioningRequest`:  
+```bash
+$ oc get provisioningrequests.o2ims.provisioning.oran.org        
+NAME      AGE   PROVISIONSTATE   PROVISIONDETAILS
+cluster-name   37m   fulfilled        Provisioning request has completed successfully
+```
+- [OADP and LCA is installed on the spoke cluster. OADP backend is configured on the spoke cluster.](https://docs.openshift.com/container-platform/4.17/edge_computing/image_based_upgrade/preparing_for_image_based_upgrade/cnf-image-based-upgrade-install-operators.html)
+```yaml
+# policytemplates/version_4.16.x/sno-ran-du/sno-ran-du-pg-v4-16-x-v1.yaml
+
+policies:
+- name: operator-subscriptions
+  manifests:
+    # LCA
+    - path: source-crs/LcaOperatorStatus.yaml
+    - path: source-crs/LcaSubscriptionNS.yaml
+    - path: source-crs/LcaSubscriptionOperGroup.yaml
+    - path: source-crs/LcaSubscription.yaml
+    # OADP
+    - path: source-crs/OadpOperatorStatus.yaml
+    - path: source-crs/OadpSubscriptionNS.yaml
+    - path: source-crs/OadpSubscriptionOperGroup.yaml
+    - path: source-crs/OadpSubscription.yaml
+- name: oadp-backend
+  manifests:
+    - path: source-crs/OadpSecret.yaml
+      patches:
+        - data: <FILL>
+    - path: source-crs/OadpDataProtectionApplication.yaml
+      patches:
+        - spec:
+            backupLocations:
+            <FILL>
+```
+### steps:
+- Create new clustertemplates and policytemplates directories for the new release according to sample git-setup.
+- Update release version in new clustertemplates files
+```yaml
+# clustertemplates/version_4.16.y/sno-ran-du/clusterinstance-defaults-v1.yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: clusterinstance-defaults-v1
+  namespace: sno-ran-du-v4-16-y
+data:
+  clusterinstance-defaults: |
+    clusterImageSetNameRef: "4.16.y"
+---
+# clustertemplates/version_4.16.y/sno-ran-du/sno-ran-du-v4-16-y-1.yaml
+kind: ClusterTemplate
+metadata:
+  name: sno-ran-du.v4-16-y-1
+  namespace: sno-ran-du-v4-16-y
+spec:
+  name: sno-ran-du
+  version: v4-16-y-1
+  release: 4.16.y
+---
+# clustertemplates/version_4.16.y/sno-ran-du/upgrade-defaults-v1.yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: upgrade-defaults-v1
+  namespace: sno-ran-du-v4-16-y
+data:
+  ibgu: |
+    ibuSpec:
+      seedImageRef:
+        image: "quay.io/seed-repo/seed-image:4.16.y"
+        version: "4.16.y"
+      oadpContent:
+        - name: sno-ran-du-ibu-platform-backup-v4-16-y-1
+          namespace: openshift-adp
+```
+- Sync ztp repo to the hub cluster.
```

Comment:
Should this be more of a "Wait for ztp repo to be synced to the hub cluster"? Won't ArgoCD do it automatically?

Reply:
I think it depends on the application settings in the ArgoCD.